core/src/test/java/org/elasticsearch/cluster/routing/OperationRoutingTests.java

```diff
@@ -75,6 +75,56 @@ public void testGenerateShardId() {
         }
     }
 
+    public void testPartitionedIndexBWC() {
```

Comment:
can we get some more basic tests here not just BWC. I'd love to see some test here that ensures we are getting the right shards back